betelgeuse.py

```diff
@@ -36,6 +37,28 @@
 
 JUNIT_TEST_STATUS = ['error', 'failure', 'skipped']
 
+# Cache for shared objects
+OBJ_CACHE = {}
+
+
+class JobNumberParamType(click.ParamType):
+    name = 'job number'
+
+    def convert(self, value, param, context):
+        if value == 'auto':
+            return multiprocessing.cpu_count()
+        try:
+            value = int(value)
+            if value <= 0:
+                raise ValueError('{0} is not a positive integer'.format(value))
+            return value
+        except ValueError:
```

Comment:
You're re-raising the error from line #53, right? If so, then you shouldn't need to create a message?